File: tests/fixtures/code-map/langs-go/defs.go
package widget

import "fmt"

const MaxSize = 100

var counter int

type Config struct {
	Name string
	size int
}

type Store interface {
	Get(k string) (string, error)
	Put(k, v string) error
}

func New(name string) *Config {
	return &Config{Name: name}
}

func (c *Config) Describe() string {
	return fmt.Sprintf("%s/%d", c.Name, c.size)
}

File: tests/fixtures/code-map/langs-rust/defs.rs
use std::fmt;

const MAX_SIZE: usize = 100;

static COUNTER: i32 = 0;

pub type NodeId = u64;

pub struct Config {
    name: String,
    size: i32,
}

pub enum Store {
    Memory,
    Disk,
}

pub trait Describe {
    fn describe(&self) -> String;
}

pub mod inner {
    pub fn helper() {}
}

pub fn new_config(name: String) -> Config {
    Config { name, size: 0 }
}

impl Config {
    pub fn describe(&self) -> String {
        let _ = fmt::Debug;
        String::new()
    }
}

macro_rules! my_macro {
    () => {};
}
